datafusion/core/Cargo.toml

```diff
@@ -80,6 +80,7 @@ ordered-float = "3.0"
 parking_lot = "0.12"
 parquet = { version = "24.0.0", features = ["arrow", "async"] }
 paste = "^1.0"
+percent-encoding = "2.2.0"
```

Comment:
`datafusion/core` uses `url` crate by servo. 
`percent-encoding` is also published by [same repository](https://github.com/servo/rust-url/tree/master/percent_encoding).